demos/shadow/shadow_demo_main/shadow_demo_helpers.c

```diff
@@ -591,25 +727,23 @@ static int waitForPacketAck( MQTTContext_t * pMqttContext,
 }
 
 /*-----------------------------------------------------------*/
-
-static MQTTStatus_t processLoopWithTimeout( MQTTContext_t * pMqttContext,
-                                            uint32_t ulTimeoutMs )
+static MQTTStatus_t processLoopWithTimeout(void)
 {
     uint32_t ulMqttProcessLoopTimeoutTime;
     uint32_t ulCurrentTime;
 
     MQTTStatus_t eMqttStatus = MQTTSuccess;
 
-    ulCurrentTime = pMqttContext->getTime();
-    ulMqttProcessLoopTimeoutTime = ulCurrentTime + ulTimeoutMs;
+    ulCurrentTime = mqttContext.getTime();
```

Comment:
Why no longer pass the context nor the timeout? This changes this removes the ability to dynamically change the timeout, or context, if a user ever wanted to reuse or extend this code.